examples/user_interfaces/toolmanager_sgskip.py

```diff
@@ -13,6 +13,10 @@
 Using `matplotlib.backend_managers.ToolManager`
 """
 
+import matplotlib
+# Not all backends have a toolbar
```

Comment:
They all have a toolbar; not all have toolmanager.
```suggestion
# Not all backends implement toolmanager.
```

Reply:
Sorry, I'm not familiar with the matplotlib terminology for these things. But what I saw was that on the `MacOSX` backend, `canvas.manager.toolmanager` actually did exist (was not `None`), and it was `canvas.manager.toolbar` that was `None`. For another example, on the `Agg` backend, both are `None` :/

Reply:
This example uses ToolManager, so `canvas.manager.toolbar` will be `None` on all backends. The `Agg` backend does not show anything, so has no toolbars of any kind.

Reply:
If `canvas.manager.toolbar` is `None`, then [this line](https://github.com/matplotlib/matplotlib/blob/3f823871f123a3d3f886c347f80addd7c21a9d2f/examples/user_interfaces/toolmanager_sgskip.py#L87) will throw an exception so the example shouldn't work at all?

Reply:
Ah sorry, I misremembered. `.toolbar` will be set to either a toolbar2 or a toolmanager, but it's _not_ the toolbar that's really missing.

Reply:
Right, so the issue is that on the `MacOSX` backend, `canvas.manager.toolbar` _was_ `None` despite a graph being shown (unlike `Agg`, for which the behaviour makes sense and I would expect the failure).

Maybe I could change the comment to something less specific, like "Tools are only available on certain backends"?